release/nightly_tests/dataset/read_benchmark.py

```diff
@@ -88,13 +97,29 @@ def run_images_benchmark(benchmark: Benchmark):
     for root in test_input:
         shutil.rmtree(root)
 
-    # TODO(chengsu): run benchmark on 20G and 100G imagenet data.
+    # Run benchmark on 1G, 20G and 100G imagenet data.
     benchmark.run(
         "images-imagenet-1g",
         read_images,
         root="s3://air-example-data-2/1G-image-data-synthetic-raw",
     )
 
+    benchmark.run(
+        "images-imagenet-20g",
+        read_images,
+        root="s3://air-example-data-2/20G-image-data-synthetic-raw",
+        # Not write to object store as the images size are too large.
+        write_to_object_store=False,
+    )
```

Comment:
I feel like not writing the images to the object store as part of the read benchmark makes the benchmark a lot less useful, since we're always going to be writing either read results or one-to-one transformation results (if stage fusion takes place) in real workloads, and the serialization overhead is certainly non-negligible.

Is there context that I'm missing here? Are we wanting to isolate the performance of read task scheduling + I/O execution from the eventual storage in Plasma for some reason?

Reply:
@clarkzinzow - yeah I agree with you here. The reason here is reading 20G, 100G images in single node (w/ no more 20G object store memory in m5.4xlarge) would cause a lot of object store spilling and failing at the end. In-memory images data size is around 8x over their on-disk size. So here take a step back, to measure the images reading performance only. We can add multi-nodes benchmark later for large datasets.

Reply:
Maybe we can tag the benchmark name to make the distinction a bit clear? like "images-imagenet-1g" is testing both read s3 and write to objectstore, whereas others are just testing read perf?

Reply:
Totally, I get the practical limitation for e.g. reading 100GB of images on a node with only 20GB object store and I understand that isolating the task scheduling + I/O execution can help us pinpoint issues there. I'm just trying to determine if this benchmark will still have any meaningful signal if the operation could never actually be completed on this cluster setup, and making sure that we're not essentially eliminating the primary system bottleneck (object store capacity) to highlight other bottlenecks that wouldn't otherwise be hit.

It seems like we're considering this benchmark to be a high-signal proxy for e.g. reading this dataset on a much larger node (whose object store could fit the entire dataset), doing a streaming read of the dataset on this node, or reading the dataset on a multi-node cluster, each of which I think we both agree would be more representative of a real-world read. So what's the motivation for adding this as a stopgap now rather than doing the higher-signal benchmark, either in this PR or later?

Reply:
Assuming the main motivation is for this to land in Ray 2.2, and us not wanting to block on creating a higher-signal benchmark, that seems fine. I certainly won't block on this, I just don't want us to forget this context and prematurely optimize the wrong bottlenecks based on this PR!

Reply:
@clarkzinzow - the main motivation is to get some nightly test coverage in single-node setup. It's still useful to catch unnecessary data buffering such as https://github.com/ray-project/ray/pull/29289. The plan to add read benchmark in multi-nodes in followup PRs. I can also remove these two benchmarks if these really bother us.

Reply:
> Maybe we can tag the benchmark name to make the distinction a bit clear? like "images-imagenet-1g" is testing both read s3 and write to objectstore, whereas others are just testing read perf?

@jianoaix - I can change the benchmark name in https://github.com/ray-project/ray/pull/30389 if it works for you. Wanted to catch today's nightly test to see if there's regression before branch cut.

Reply:
@clarkzinzow, @jianoaix - based on the feedback, removing the read benchmark for 20G, 100G for now. Will add in multi-nodes benchmark later. Thanks!